src/builtins.ts

```diff
@@ -113,11 +114,24 @@ function any(x: collections.Iterable): boolean {
 }
 
 function bool(x) {
-    if (x === undefined)
+    
+    if(x instanceof Py_Tuple || x instanceof iterator.XRange || x instanceof pytypes.Py_Str 
+       || x instanceof Py_List || x instanceof Py_Dict){
+        return x.len() !== 0;
+    }
+    if(x instanceof Py_Int || x instanceof Py_Float){
+        return x.toNumber() != 0;
+    }
+    if(x instanceof Py_Long){
+        return !x.eq(Py_Long.fromInt(0));
+    }
+    if(x instanceof Py_Complex){
+        return !(x.real.value == 0 && x.imag.value == 0);
+    }
+    if(x == singletons.None || x == false){
         return false;
-    if (x['bool'] !== undefined)
-        return x.bool();
-    return x != 0;
+    }
+    return true;
```

Comment:
You could actually just return the negation of the above conditional that returns `false`:

```
return !(x == singletons.None || x == false);
```

which, by de Morgan's law, is equivalent to:

```
return x != singletons.None && x != false;
```